### DISCUSSION TITLE
Chakra UI flashes wrong UI on load.

### DISCUSSION THREAD

**Discussion by AnonymousXC** [chakra-ui/chakra-ui#7259] [Q&A] [answered]
Hi,
I am having a full loading screen ( width : 100% and height : 100%). 
And, I want it to display first then when the UI is ready after useMediaQuery() hooks are resolved.
Then, I want to hide it.
But, Chakra UI flashes the wrong UI. That is, first the whole unloaded content and then the loading screen. And, it looks very ugly.
Using NextJS 13.
Thanks for help.

**Comment by ArthurPedroti** [comment]
Hey @AnonymousXC, do you have some print screens to show the behavior?

**Reply by AnonymousXC** [reply]
https://user-images.githubusercontent.com/96725322/213427596-607597b0-e870-48f8-98ff-1b10f69944be.mp4

Like, here the site flashes first then, the loading screen is seen. 🙏

**Reply by ArthurPedroti** [reply]
Do you have the repo or the code of your component/page to show to us?

**Reply by AnonymousXC** [reply]
I have the source code [here](https://github.com/AnonymousXC/freelance).

I have not **rendered the `LoadingScreen` component** as I wanted a working version of site on the repo.
But, I have **imported the loadingscreen component in `_app.tsx` file but I have not put it inside the `ChakraProvider`.**

**Comment by ArthurPedroti** [comment, marked as answer]
You are using Next.js, why don't use server-side rendering with static pages to don't need to load anything?

**Reply by AnonymousXC** [reply]
Thanks.

After hearing this, I thought to generate static HTML of the page but I did not worked at all, specially theme change button.

So, I import all the components as :-

`import dynamic from 'next/dynamic';`

`const component = dynamic(() => import('...path'), { ssr : false})`

**Reply by ArthurPedroti** [reply]
Next.js have many different ways to handle different types of pages, with new features of react you can also only re-render some isolated components of your UI, and you have to look better what is the better way to handle themes in each UI library you want to use, because each UI library handles the theme changes in different ways.

Is always better to try to understand the way the creator of the library wants to handle different approaches, and if this way fits with other tools and libraries you are using. Because sometimes, the engineering you have to do to "fit" only one library on another is not a good trade-off and you will have more headaches in the future just to keep that "engineering" over the many updates that different libraries have over time.

**Reply by AnonymousXC** [reply]
Very helpful. TY. 😆
